engine/src/io/data_parser/sql/MySQLParser.cpp

```diff
@@ -0,0 +1,598 @@
+  if (StringUtil::beginsWith(t, "BOOL") ||
+      StringUtil::beginsWith(t, "BOOLEAN") || (t == "TINYINT(1)")) return cudf::type_id::BOOL8;
+  if (StringUtil::beginsWith(t, "TINYINT")) return cudf::type_id::INT8;
+  if (StringUtil::beginsWith(t, "INT") ||
+      StringUtil::beginsWith(t, "INTEGER")) return cudf::type_id::INT32;
+  if (StringUtil::beginsWith(t, "BIGINT")) return cudf::type_id::INT64;
+  if (StringUtil::beginsWith(t, "FLOAT")) return cudf::type_id::FLOAT32;
+  if (StringUtil::beginsWith(t, "DOUBLE")) return cudf::type_id::FLOAT64;
+  // test date/datetime data types ...
+  if (t == "DATE") return cudf::type_id::TIMESTAMP_DAYS;
+  if (t == "TIME") return cudf::type_id::TIMESTAMP_SECONDS; // without the date part
+  if (StringUtil::beginsWith(t, "DATETIME")) return cudf::type_id::TIMESTAMP_MILLISECONDS;
+  if (StringUtil::beginsWith(t, "TIMESTAMP")) return cudf::type_id::TIMESTAMP_MILLISECONDS;
+  if (StringUtil::beginsWith(t, "YEAR")) return cudf::type_id::INT8;
+  // decimal TODO percy william c.cordova
+  if (StringUtil::beginsWith(t, "DECIMAL")) return cudf::type_id::FLOAT64;
+
+  // TODO percy ...
+}
+
+std::vector<cudf::type_id> parse_mysql_column_types(const std::vector<std::string> types) {
+  std::vector<cudf::type_id> ret;
+  for (auto t : types) {
+    ret.push_back(parse_mysql_column_type(t));
+  }
+  return ret;
+}
+
+cudf::io::table_with_metadata read_mysql(std::shared_ptr<sql::ResultSet> res,
+                                         const std::vector<int> &column_indices,
+                                         const std::vector<cudf::type_id> &cudf_types,
+                                         size_t total_rows)
+{
+  cudf::io::table_with_metadata ret;
+  std::vector<void*> host_cols(column_indices.size());
+  auto const num_words = cudf::bitmask_allocation_size_bytes(total_rows)/sizeof(cudf::bitmask_type);
+  auto null_masks = std::vector<std::vector<cudf::bitmask_type>>(host_cols.size());
+
+  for (int col = 0; col < host_cols.size(); ++col) {
+    null_masks[col].resize(num_words, 0);
+    size_t projection = column_indices[col];
+    cudf::type_id cudf_type_id = cudf_types[projection];
+    switch (cudf_type_id) {
+      case cudf::type_id::EMPTY: {} break;
+      case cudf::type_id::INT8: {
+        std::vector<int8_t> *tmp = new std::vector<int8_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::INT16: {
+        std::vector<int16_t> *tmp = new std::vector<int16_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::INT32: {
+        std::vector<int32_t> *tmp = new std::vector<int32_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::INT64: {
+        std::vector<int64_t> *tmp = new std::vector<int64_t>();
+        tmp->resize(total_rows);
+        host_cols[col] = tmp;
+      } break;
+      case cudf::type_id::UINT8: {
```

Comment:
Hmmmm, have you tried Unsigned integers? BSQL does not support unsigned integers yet. Not sure what would happen if you tried using them here.  I would guess we should leave this part here as is, and in the schema parsing, make sure that it maps to a signed integer

Reply:
what we did before was make the smller types larger. though now that i mention it this is just the allocation logic basically. What is the purpose of allocating a std vector then casting it to void? Wouldnt it be the same to just allocate the space?

Reply:
I would recommend making `host_cols` a `std::vector<std::vector<char>>` so that you can just set the size of each `std::vector<char>` the size in bytes that it needs to hold the data and you don't need to do `new` which can lead to memory leaks if not handled correctly. You can then do a reinterpret_cast when setting or getting the values.

Reply:
@williamBlazing that was the first implementation of `host_cols` and if we do that we need to iterate all the rows again to construct the buffer for the cudf column. The way is implemented right now is pretty good because we save like 2 loops over all the rows and the data structs are exactly what the cudf ctors want cc @felipeblazing @gcca

Reply:
Perhaps I am not explaining myself. Right now you are holding the data in vectors of specific types that you are creating with `new`, but then you are type-erasing them and holding them as `void*` pointers in `host_cols`. My suggestion was that you replace the individual typed vectors you are creating with `new` and only use type erased vectors of type char that you dont have to use `new`. You can do a reinterpret_cast on those vectors of char to give them a type when you want to put a value or create a cudf from them. The purpose of this suggestion is so that you dont have to use `new` which can lead to memory leaks if not handled correctly

Reply:
So the issue is about code style I guess, the `new` is being used but we are also using the `delete` to avoid any leaks.
The approach you are suggesting was the first implementation and it didn't work, because some types cannot be casted even using `reinterpred_cast`, this will fails for long double and strings.
If the core issue is about the use of `new` then note that we just need to make sure we are deleting correctly, and that is the case here. cc @felipeblazing @gcca

Reply:
hes suggestiong you use a std::vector<char> no new and delete needed but it has another issue. the <char> there has no guaranteed alignment so however you allocate that vector you need to make sure its aligned to the largest type you want to represent.

Reply:
I agree that the whole type erasing a vector of something only to reinterpret cast it later is more problematic than the need to manage the memory resource.. it should be at least 8 byte aligned on a 64 bit system if im not mistaken so until we have 128 bit integers it might not matter

Reply:
great thanks, yeah and we are taking care to manage the memory resource there in the abstract_sql_parser, soon in the next PR postgresql and sqlite will use that same code so there will be only a single place where we are using new and delete.

Reply:
One solution we can try in the future is to use std::shared_ptr<void> instead of a naked void*, we can use std::unique_ptr here but that would require mor code to create custom dtors (std::shared_ptr infers the type for us) 
However I think dealing with a better style/approach for this code will be subject for another PR, the way it is now do the job and doesn't leak